app/docs/src/how-tos/create-templates.md

```diff
@@ -0,0 +1,551 @@
+
+## Common Patterns
+
+The walkthrough is a great place to get started, but the following common patterns can help supercharge your templates.
+
+### Setting Subscriptions
+
+Connect components by setting subscription attributes:
+
+```typescript
+// By component name
+await ctx.ensureAttribute(
+  subnet,
+  "/domain/extra/Region",
+  {
+    kind: "$source",
+    component: "us-east-1",
+    path: "/domain/region",
+  }
+);
+
+// By search query
+await ctx.ensureAttribute(
+  subnet,
+  "/domain/extra/Region",
+  {
+    kind: "search",
+    query: 'name:"us-east-1"',
+    path: "/domain/region",
+  }
+);
+```
+
+### Creating New Components
+
+Add new components to the working set:
+
+```typescript
+ctx.transform(async (workingSet, inputs) => {
+  const newSubnet = await ctx.newComponent(
+    "AWS::EC2::Subnet",
+    "additional-subnet",
+    {
+      "/domain/CidrBlock": "10.0.3.0/24",
+      "/domain/AvailabilityZone": "us-east-1c",
+    }
+  );
+  workingSet.push(newSubnet);
+
+  return workingSet;
+});
+```
+
+### Copying Components
+
+Create multiple similar components from an example:
+
+```typescript
+ctx.transform(async (workingSet, inputs) => {
+  const template = workingSet.find((c) => c.name.includes("template"));
+
+  if (template) {
+    for (let i = 1; i <= 3; i++) {
+      const copy = ctx.copyComponent(template, `server-${i}`);
+      await ctx.ensureAttribute(
+        copy,
+        "/domain/InstanceType",
+        "t3.medium"
+      );
+      workingSet.push(copy);
+    }
+  }
+
+  return workingSet;
+});
+```
+
+### Using `SubscriptionInput` for Component References
+
+See [Add External Component References with SubscriptionInput](#add-external-component-references-with-subscriptioninput) in the main walkthrough for a complete example.
+
+You can reference components in two ways within input files:
+
+1. By search query (recommended for finding components by schema or tags):
+
+```yaml
+credential:
+  kind: "search"
+  query: 'schema:"AWS Credential"'
+  path: "/secrets/AWS Credential"
+```
+
+2. By component name (when you know the exact component name)
+
+```yaml
+region:
+  kind: "$source"
+  component: "us-east-1"
+  path: "/domain/region"
+```
+
+## Best Practices
+
+Apart from [Common Patterns](#common-patterns), there are some ways to make creating templates a smooth experience.
+
+### Start with Working Infrastructure
+
+You can [create](../tutorials/creating-infrastructure.md) and [discover](../tutorials/discover.md) infrastructure before templating it. This ensures that you start with a desired baseline before templating.
+
+### Version Your Baselines
+
+You can commit baseline caches alongside templates in source control. This creates a version history of your infrastructure patterns:
+
+```
+templates/
+├── aws-standard-vpc.ts
+├── aws-standard-vpc-baseline.yaml
+└── aws-standard-vpc-inputs.yaml
+```
+
+### Test with Dry Runs
```

Comment:
This probably should be added to the refference docs, since I don't think I covered it

Reply:
I'll add it there